webapp/src/Controller/Jury/SubmissionController.php

```diff
@@ -390,7 +390,7 @@ public function viewAction(
                     $runResult['hostname'] = $firstJudgingRun->getJudgeTask()->getJudgehost()->getHostname();
                     $runResult['judgehostid'] = $firstJudgingRun->getJudgeTask()->getJudgehost()->getJudgehostid();
                 }
-                $runResult['is_output_run_truncated'] = preg_match(
+                $runResult['is_output_run_truncated'] = $outputDisplayLimit >= 0 && preg_match(
```

Comment:
Is this correct? My understanding is that `$runResult['is_output_run_truncated']` indicates whether the final displayed output is truncated or not. This could have happened at storage and/or fetch time (above). If `$outputDisplayLimit == -1`, but output was truncated at storage time, then this seems wrong.
I think the simplest and clearest would be something like:
```php
                preg_match(
                    '/\[output storage truncated after \d* B\]/',
                    (string)$runResult['output_run_last_bytes']
                ) ||
                preg_match(
                    '/\[output display truncated after \d* B\]/',
                    (string)$runResult['output_run']
                );
```

As a side note: I can't find `TRUNCATE` from https://github.com/DOMjudge/domjudge/pull/2379/files#diff-9f6bb8e6b77ae7a86ba0f4456c3803399e916c0979d8c742dfa7253987e55423R340-R346 above as a MySQL string function, only https://dev.mysql.com/doc/refman/8.0/en/mathematical-functions.html#function_truncate. Does anyone know where this comes from?

Reply:
I sent a PR to fix this https://github.com/DOMjudge/domjudge/pull/2410